src/AggregateFunctions/AggregateFunctionOrderBy.h

```diff
@@ -0,0 +1,162 @@
+#pragma once
+
+#include <AggregateFunctions/IAggregateFunction.h>
+#include <Columns/ColumnArray.h>
+#include <Common/ArenaAllocator.h>
+#include <Common/assert_cast.h>
+#include <DataTypes/DataTypeArray.h>
+#include <Interpreters/AggregationCommon.h>
+
+namespace DB
+{
+
+    namespace ErrorCodes
+    {
+        extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+    }
+
+struct AggregateFunctionOrderByData
+{
+    AggregateFunctionOrderByData() = default;
+    explicit AggregateFunctionOrderByData(size_t num_arguments): value(num_arguments), arena(nullptr) {}
+
+    using Array = std::vector<IColumn*>;
+
+    Array value;
+    std::once_flag initialized;
```

Comment:
The proposed variant of implementation is not the correct way to implement this task.